brainglobe_heatmap/heatmaps.py

```diff
@@ -48,6 +50,187 @@ def check_values(values: dict, atlas: Atlas) -> Tuple[float, float]:
+      2. For each candidate point, compute the distance to the
+         closest polygon edge if the point is inside the polygon.
+      3. Use a priority queue (max-heap) that have the
+         greatest potential to contain a point with a larger distance.
+      4. Repeatedly extract the cell with the
+         largest potential distance from the queue:
+      5. Continue until no cells can offer an improvement within precision.
+
+    Parameters
+    ----------
+    polygon_vertices : np.ndarray of shape (N, 2)
+        Array of (x, y) coordinates defining the polygon vertices.
+    precision : float, optional
+        The precision tolerance for the result (default: 1.0).
+        A smaller value yields higher accuracy.
+
+    Returns
+    -------
+    tuple
+        The (x, y) coordinates of the optimal annotation position.
+    """
+
+    def calculate_point_to_edges_distance(
+        px: float, py: float, vertices: np.ndarray
+    ) -> float:
+        """
+        Calculate the minimum distance from a point to the edges of a polygon.
+
+        Returns
+        -------
+        float
+            Minimum distance from (x, y) to any edge if inside the polygon,
+            float('-inf') if outside the polygon.
+        """
+        point = np.array([px, py])
+
+        if not polygon.contains_point((px, py)):
+            return float("-inf")
+
+        segments = np.vstack((vertices, vertices[0]))
+        segment_starts = segments[:-1]
+        segment_ends = segments[1:]
+
+        edges = segment_ends - segment_starts
+        lengths_sq = np.sum(edges**2, axis=1)
+        valid_edges = lengths_sq > 0
+        min_dist = float("inf")
+
+        # find the minimum distance from a point to a line segment
+        if np.any(valid_edges):
+            segment_ratio = (
+                np.sum(
+                    (point - segment_starts[valid_edges]) * edges[valid_edges],
+                    axis=1,
+                )
+                / lengths_sq[valid_edges]
+            )
+            segment_ratio = np.clip(segment_ratio, 0, 1)
+
+            closest = (
+                segment_starts[valid_edges]
+                + (edges[valid_edges].T * segment_ratio).T
+            )
+            distances = np.sqrt(np.sum((point - closest) ** 2, axis=1))
+            min_dist = min(min_dist, np.min(distances))
+
+        if np.any(~valid_edges):
+            closest = segment_starts[~valid_edges]
+            vertex_distances = np.sqrt(np.sum((point - closest) ** 2, axis=1))
+            if len(vertex_distances) > 0:
+                min_dist = min(min_dist, np.min(vertex_distances))
+
+        return min_dist
+
+    if not isinstance(polygon_vertices, np.ndarray):
+        polygon_vertices = np.asarray(polygon_vertices)
+
+    polygon = pltpath.Path(polygon_vertices)
+
+    # determine the bounding box of the polygon
+    minx, miny = np.min(polygon_vertices, axis=0)
+    maxx, maxy = np.max(polygon_vertices, axis=0)
```

Comment:
```suggestion
    minx = np.min(polygon_vertices, axis=0)
    miny = np.min(polygon_vertices, axis=1)
```
I would naively think?
Is this currently a bug, or do I need an explanation of the `pltpath.Path` stores polygonal vertices? :smile:

Reply:
It is working as intended, looking at this snippet I reckon it would be a good idea to change variable naming to min_x min_y and max_x min_y to be more pythonic. I will take a look.

I will try to explain,
_I am forcing polygon_vertices to numpy.ndarray(np.ndarray)._
When you do, 
- np.min(polygon_vertices, axis=0) will give the minimum value from each column, good for our needs, one array with [minx, miny].
- When you do, np.min(polygon_vertices, axis=1) will give the minimum value within each row, one array with one value per row.

```python
test = [
   [1, 2],
   [3, 4],
   [5, 6],
   [6, 1]
]
array_with_min_x_and_miny = np.min(test, axis=0) # [1 1]
array_with_min_values_of_each_vertice = np.min(test, axis=1) # [1 3 5 1]
```

Reply:
I see, all clear - thanks for the explanation!